### PR TITLE
Fixed Pre-Build Event Issues and Bootstrap / Resource Gen Errors

### PR DESCRIPTION
**Pull request by jbakert**
Several batch files would throw errors in the original solution and cause problems when the HSDT projects were included in a 3rd party solution (usually a plugin). 

Added error checking to scripts to prevent execution of routines that were contingent upon a successful completion of the previous command.  

Added the ability to disable the PreBuildEvent if you so desired. Fixed missing nuget.exe for VS2017 by using msbuild's new integrated functionality. 

Also rocked a nice ^M in there somewhere.

### CODE REVIEW

File: Hearthstone Deck Tracker/Hearthstone Deck Tracker.csproj [hunk 1]
```diff
@@ -1398,36 +1400,9 @@
...
-) else (
-  git clone --depth 1 https://github.com/HearthSim/HDT-Localization.git $(SolutionDir)HDT-Localization
-)
-xcopy /Y "$(SolutionDir)HDT-Localization\*.resx" "$(ProjectDir)Properties\"
-mkdir "$(TargetDir)\Images\Tiles"
-mkdir "$(TargetDir)\Images\Themes"
-xcopy /E /Y /Q "$(SolutionDir)Resources\Generated\Tiles" "$(TargetDir)\Images\Tiles"
-xcopy /E /Y /Q "$(ProjectDir)Images\Themes" "$(TargetDir)\Images\Themes"
-xcopy /Y /Q "$(SolutionDir)CHANGELOG.md" "$(ProjectDir)Resources"</PreBuildEvent>
+    <PreBuildEvent>"$(ProjectDir)PreBuildEvent.bat" "$(ProjectDir)..\" "$(ProjectDir)" "$(ProjectDir)($OutDir)" 0 0
```
Comment (azeier) [inline Hearthstone Deck Tracker/Hearthstone Deck Tracker.csproj]: This should probably be `$(OutDir)`? Also, why not keep `$(TargetDir)`?

File: bootstrap.bat [hunk 2]
```diff
@@ -1,4 +1,7 @@
-nuget restore
+@echo off
+
+msbuild /t:restore
```
Comment (azeier) [inline bootstrap.bat]: This doesn't seem to work for me. I get the same error as [AppVeyor](https://ci.appveyor.com/project/azeier/hearthstone-deck-tracker) when running bootstrap locally on a clean clone.

File: bootstrap.bat [hunk 3]
```diff
@@ -1,4 +1,7 @@
-nuget restore
+@echo off
+
+msbuild /t:restore
+REM nuget restore
```
Comment (azeier) [inline bootstrap.bat]: Assuming the line above works, this can probably be removed.

File: Hearthstone Deck Tracker/Test.bat [hunk 4]
```diff
@@ -0,0 +1,3 @@
+@echo off
```
Comment (azeier) [inline Hearthstone Deck Tracker/Test.bat]: This file should probably be removed.

File: bootstrap.bat (hunk 2 above)
Comment (jbakert) [inline bootstrap.bat]: I noticed a lot of nuget references in your output. Is it possible that the version of nuget you're using is incompatible? When looking for a nuget for 2017, I came across an article that stated the command line nuget was deprecated and being phased out by being integrated into msbuild

File: bootstrap.bat (hunk 3 above)
Comment (jbakert) [inline bootstrap.bat]: The REM?  Yes, I agree.

File: Hearthstone Deck Tracker/Test.bat (hunk 4 above)
Comment (jbakert) [inline Hearthstone Deck Tracker/Test.bat]: Yes

File: bootstrap.bat (hunk 2 above)
Comment (jbakert) [inline bootstrap.bat]: Welp. I was playing with .NET Core. I've reverted the change. Would it be prudent to include the nuget 4.1 binary in the root solution folder?

Comment (judge2020) [inline bootstrap.bat]: It's usually recommended to have Nuget in your path somewhere.

File: Hearthstone Deck Tracker/Hearthstone Deck Tracker.csproj (hunk 1 above)
Comment (jbakert) [inline Hearthstone Deck Tracker/Hearthstone Deck Tracker.csproj]: Yes, I made this change.

File: bootstrap.bat (hunk 2 above)
Comment (jbakert) [inline bootstrap.bat]: Fair. But path parsing order, multiple scattered versions, and explicit binaries necessary to get a solution in working order mimic the package system, so I think the addition of this tool to be prudent. That being said, I don't mind either way.

**Comment by azeier** [comment]
Not a fan of the submodules or including nuget.exe. Otherwise this looks gtm.

**Comment by judge2020** [comment]
^ 
Most of the time, it is recommended to have nuget somewhere (else) in your path. It's easy to download it from nuget.org anyways.

**Pull request merged**